clusterloader/cluster_loader.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cl
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	clframe "github.com/sjug/perf-tests/clusterloader/framework"
+	"k8s.io/kubernetes/pkg/api/v1"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = framework.KubeDescribe("Cluster Loader [Performance] [Slow] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("cluster-loader")
+	defer GinkgoRecover()
+
+	var c clientset.Interface
+	BeforeEach(func() {
+		c = f.ClientSet
+	})
+
+	//It(fmt.Sprintf("running config file: %v, length: %v, type %T", project, len(project), project), func() {
+	It(fmt.Sprintf("running config file"), func() {
+		project := clframe.ConfigContext.ClusterLoader.Projects
+		tuningSets := clframe.ConfigContext.ClusterLoader.TuningSets
+		if len(project) < 1 {
+			framework.Failf("invalid config file.\nFile: %v", project)
+		}
+
+		var namespaces []*v1.Namespace
+		//totalPods := 0 // Keep track of how many pods for stepping
+		for _, p := range project {
+			// Find tuning if we have it
+			tuning := getTuningSet(tuningSets, p.Tuning)
+			framework.Logf("Our tuning set is: %v", tuning)
+			for j := 0; j < p.Number; j++ {
+				// Create namespaces as defined in Cluster Loader config
+				nsName := p.Basename + strconv.Itoa(j)
```

Comment:
I don't know what this means, how would I use a struct to improve this?
Make a helper fn? `func appendIntToString(i int) string`?